shotover/src/hot_reload/server.rs

```diff
@@ -184,15 +185,57 @@ pub fn start_hot_reload_server(socket_path: String, sources: &[Source]) {
         .collect();
 
     tokio::spawn(async move {
-        match UnixSocketServer::new(socket_path, source_handles) {
-            Ok(mut server) => {
-                info!("Unix socket server started for hot reload communication");
-                if let Err(e) = server.run().await {
-                    error!("Unix socket server error: {:?}", e);
+        const MAX_RETRIES: u32 = 100;
```

Comment:
There should be no maximum retries, shotover should continue retrying forever until this succeeds.
Usually a max retries is a good idea, but in this case it is critical that this succeeds eventually and there is only one instance of the loop so it wont impact performance.